### ISSUE TITLE
Add a "list templates" action to actors and apps init workflows

### ISSUE THREAD

**Issue by mwvaughn** [TACC-Cloud/tapis-cli#351]
**Problem**
It is not apparent to end users what templates are available to choose from when running one of the `init` workflows. 

**Proposed Solution**
1. Implement a `-L, --list-templates` workflow flag in both `init` workflow commands. When this code path is called, the CLI will fetch a structured file from the CookieCutter repository then render it as a table for the end user to view. No other workflow actions will be taken if this flag is specified. 

**Notes**
An example structured file might look like this:

_catalog.json_
```
{
    "default": {"description": "Description 1 64 chars"},
    "shellrun": {"description": "Description 2 64 chars"},
    "wordcount": {"description": "Description 3 64 chars"}
}
```

It would be available at the root level of the CookieCutter repository and thus retrievable (if hosted on Github) using raw.githubusercontent.com instead of the more formal Git API. For example, this file stored at `TACC-Cloud/cc-tapis-v2-app/catalog.json` would be accessible over plain HTTP/S at https://raw.githubusercontent.com/TACC-Cloud/cc-tapis-v2-app/master/catalog.json

The primary key for the _catalog.json_ document will be the name of the subdirectory where a cookiecutter template is found. Additional metadata such as `description` will be held in a JSON dict as the value. This extensible structure will allow us to add more metadata such as `"level": "beginner"` and render it in the CLI. 

We will need to coordinate work with https://github.com/TACC-Cloud/cc-tapis-v2-app and https://github.com/TACC-Cloud/cc-tapis-v2-actors repositories to make this work.

**Comment by mwvaughn** [comment]
I ended up adding this file to the current apps library repo:

```
{
	"default": {
		"name": "Default",
		"description": "Basic code and configuration skeleton",
		"level": "beginner"
	},
	"shellrun": {
		"name": "Shell Runner",
		"description": "Run an arbitrary shell command via Tapis",
		"level": "beginner"
	},
	"wordcount": {
		"name": "Word Count",
		"description": "Simple word counting implementation",
		"level": "beginner"
	}
}
```

**Comment by mwvaughn** [comment]
This is complete and merged in as of 18e0b5c9fbc1efb469e229ee074d5df4cb752d3c
